File: AndroidNative/List/app/src/main/java/dfc/performancetest/CustomListAdapter.java
package dfc.performancetest;

import android.app.Activity;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.net.Uri;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.squareup.picasso.Picasso;

import java.io.BufferedInputStream;
import java.io.IOException;
import java.io.InputStream;
import java.net.HttpURLConnection;
import java.net.MalformedURLException;
import java.net.URI;
import java.net.URL;
import java.net.URLConnection;

/**
 * Created by diego.castro on 26/09/2016.
 */
public class CustomListAdapter extends ArrayAdapter<String> {

    private final Activity context;
    private final String[] itemname;
    private final String[] imgSource;

    public CustomListAdapter(Activity context, String[] itemname, String[] imgSource) {
        super(context, R.layout.list, itemname);
        // TODO Auto-generated constructor stub

        this.context=context;
        this.itemname=itemname;
        this.imgSource=imgSource;
    }

    public View getView(int position, View view, ViewGroup parent) {
        LayoutInflater inflater=context.getLayoutInflater();
        View rowView=inflater.inflate(R.layout.list, null,true);

        TextView txtTitle = (TextView) rowView.findViewById(R.id.Itemname);
        ImageView imageView = (ImageView) rowView.findViewById(R.id.ItemImage);

        txtTitle.setText(itemname[position]);

        Picasso.with(context).load(imgSource[position]).into(imageView);

        return rowView;
    }

    private Bitmap getImageBitmap(String url) {
        Bitmap bm = null;
        try {
            URL aURL = new URL(url);
            URLConnection conn = aURL.openConnection();
            conn.connect();
            InputStream is = conn.getInputStream();
            BufferedInputStream bis = new BufferedInputStream(is);
            bm = BitmapFactory.decodeStream(bis);
            bis.close();
            is.close();
        } catch (IOException e) {
            Log.e("dfc", "Error getting bitmap", e);
        }
        return bm;
    }
}

File: XamarinAndroid/List/CustomListAdapter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using Square.Picasso;

namespace PerformanceTest
{
    public class CustomListAdapter : BaseAdapter<string>
    {
        private Activity context;
        private String[] itemname;
        private String[] imgSource;

        public override int Count
        {
            get
            {
                return itemname.Length;
            }
        }

        public override string this[int position]
        {
            get
            {
                return itemname[position];
            }
        }

        public CustomListAdapter(Activity context, int textViewResourceId, String[] itemname, String[] imgSource) : base()  
        {
            this.context = context;
            this.itemname = itemname;
            this.imgSource = imgSource;           
        }

        public override View GetView(int position, View convertView, ViewGroup parent)
        {
            LayoutInflater inflater = context.LayoutInflater;
            View rowView = inflater.Inflate(Resource.Layout.list, null, true);

            TextView txtTitle = (TextView)rowView.FindViewById(Resource.Id.Itemname);
            ImageView imageView = (ImageView)rowView.FindViewById(Resource.Id.ItemImage);            

            txtTitle.Text = itemname[position];            

            Picasso.With(context).Load(imgSource[position]).Into(imageView);

            return rowView;
        }

        public override long GetItemId(int position)
        {
            return position;
        }        
    }
}